src/env/real.rs

```diff
@@ -0,0 +1,38 @@
+use std::ffi::OsString;
+use std::fmt;
+use std::io;
+
+use super::Env;
+
+/// Use real I/O
+pub struct RealEnv;
+
+impl Env for RealEnv {
+    // fn make_connection(&self) {
```

Comment:
As you don't define the commented out `stdin()` from the trait here you could also omit these commented out functions everywhere for the moment except in the trait itself.

Reply:
Done!